tasks/quality.js

```diff
@@ -25,31 +25,38 @@ module.exports = function(grunt) {
   // Task set aliases are registered at the end of the file based on these values.
   var validate = [];
   var analyze = [];
+  var themes = grunt.config('config.themes');
 
   var defaultPatterns = [
-    '<%= config.srcPaths.drupal %>/**/*.php',
-    '<%= config.srcPaths.drupal %>/**/*.module',
-    '<%= config.srcPaths.drupal %>/**/*.inc',
-    '<%= config.srcPaths.drupal %>/**/*.install',
-    '<%= config.srcPaths.drupal %>/**/*.profile',
-    '!<%= config.srcPaths.drupal %>/sites/**',
-    '!<%= config.srcPaths.drupal %>/**/*.box.inc',
-    '!<%= config.srcPaths.drupal %>/**/*.features.*inc',
-    '!<%= config.srcPaths.drupal %>/**/*.pages_default.inc',
-    '!<%= config.srcPaths.drupal %>/**/*.panelizer.inc',
-    '!<%= config.srcPaths.drupal %>/**/*.strongarm.inc'
+    '<%= config.srcPaths.drupal %>/{modules,profiles,libraries,static}/**/*.{php,module,inc,install,profile}',
+    '!<%= config.srcPaths.drupal %>/{modules,profiles,libraries,static}/**/*.{box,pages_default,views_default,panelizer,strongarm}.inc',
+    '!<%= config.srcPaths.drupal %>/{modules,profiles,libraries,static}/**/*.features.*inc',
+    '!<%= config.srcPaths.drupal %>/{modules,profiles,libraries,static}/**/vendor/**'
   ];
 
+  // Include common sites and theme locations in phplint validation.
+  var phplintPatterns = defaultPatterns.slice(0);
+  phplintPatterns.unshift('<%= config.srcPaths.drupal %>/sites/**/*.{php,inc}');
+
+  for (var key in themes) {
```

Comment:
If a theme is not wired into the GDT config, we still want to scan it. The theme configuration is intended to target themes taking over more of their own functionality.

Maybe this works as:
```
     '<%= config.srcPaths.drupal %>/themes/*/template.php',
     '<%= config.srcPaths.drupal %>/themes/*/templates/**/*.php',
     '<%= config.srcPaths.drupal %>/themes/*/includes/**/*.{inc,php}'
```